src/H5FDonion.c

```diff
@@ -772,6 +772,126 @@ H5FD__onion_create_truncate_onion(H5FD_onion_t *file, const char *filename, cons
     FUNC_LEAVE_NOAPI(ret_value);
 } /* end H5FD__onion_create_truncate_onion() */
 
+static herr_t
+H5FD__onion_remove_unused_symbols(char *s)
```

Comment:
Just tokenizing with strtok on {, }, ' ' symbols is probably a little more elegant, but not a big deal. However, since the config string was retrieved with H5P_peek_driver_config_str, this will probably directly modify it, which isn't ideal.